### PR TITLE
Ignore module phase and add nice debugging option

### PR DESCRIPTION
**Pull request by davidot**
See https://github.com/tc39/test262/issues/3647

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -481,6 +481,25 @@ static bool verify_test(Result<void, TestError>& result, TestMetadata const& met
 
     got_error = JsonValue { error_to_json(error) };
 
+    if (metadata.program_type == JS::Program::Type::Module && metadata.type == "SyntaxError"sv) {
+        // NOTE: Since the "phase" of negative results is both not defined and hard to
+        //       determine track throughout the entire Module life span
```
Comment (linusg) [inline src/main.cpp:486]: "hard to determine track" - missing word?

**Review by linusg — COMMENTED** [review]
That `-d` option is :yakkiss:

File: src/main.cpp [hunk 2]
```diff
@@ -517,13 +536,15 @@ void __assert_fail(const char* assertion, const char* file, unsigned int line, c
...
     int timeout = 10;
+    bool enable_debug_printing = false;
 
     Core::ArgsParser args_parser;
     args_parser.set_general_help("LibJS test262 runner for streaming tests");
     args_parser.add_option(s_harness_file_directory, "Directory containing the harness files", "harness-location", 'l', "harness-files");
     args_parser.add_option(s_use_bytecode, "Use the bytecode interpreter", "use-bytecode", 'b');
     args_parser.add_option(s_parse_only, "Only parse the files", "parse-only", 'p');
     args_parser.add_option(timeout, "Seconds before test should timeout", "timeout", 't', "seconds");
+    args_parser.add_option(enable_debug_printing, "Enable debug printing", "debug", 'd');
```
Comment (linusg) [inline src/main.cpp:547]: Hold on, this is in the runner directly... how does the `main.py` script interface with this option, i.e. how do you use it?

Comment (davidot) [reply src/main.cpp:547]: I use it when running the runner directly, I can add an option to the main.py script but that probably gets quite spammy. Should I add an option there as well?

Comment (davidot) [reply src/main.cpp:547]: Done

File: main.py [hunk 3]
```diff
@@ -504,7 +506,7 @@ def main() -> None:
     parser.add_argument(
         "-j",
         "--libjs-test262-runner",
-        required=True,
+        default="./Build/libjs-test262-runner",
```
Comment (davidot) [inline main.py:509]: Also added this, which is a super old commit I had lying around. Let me know if you disagree with this change I can just remove it again.

Comment (linusg) [reply main.py:509]: Nope, that's great :+1:

**Comment by davidot** [comment]
Using `--debug` with `main.py` is a little bit ugly.
This is with a dbgln right after starting up:
```
Process did not fail but still there is stderr output:                                                                                      
printing!

Process did not fail but still there is stderr output:                                                                                      
printing!

```

But I'll leave it like that for now, anyone who uses it can improve it later :^).

File: main.py [hunk 4]
```diff
@@ -88,7 +88,7 @@ def limit_memory():
     command = [
         str(libjs_test262_runner),
         *(["-b"] if use_bytecode else []),
-        *(["--parse-only"] if parse_only else []),
+        *(extra_runner_options),
```
Comment (linusg) [inline main.py:91]:
```suggestion
        *extra_runner_options,
```

File: main.py [hunk 5]
```diff
@@ -256,7 +256,7 @@ def __init__(
         use_bytecode: bool = False,
         track_per_file: bool = False,
         fail_only: bool = False,
-        parse_only: bool = False,
+        extra_runner_options: list[str] = None,
```
Comment (linusg) [inline main.py:259]:
```suggestion
        extra_runner_options: list[str] | None = None,
```

File: main.py [hunk 6]
```diff
@@ -275,7 +275,9 @@ def __init__(
         self.file_result_map: dict[str, str] = {}
         self.total_count = 0
         self.duration = datetime.timedelta()
-        self.parse_only = parse_only
+        self.extra_runner_options = (
+            [] if extra_runner_options is None else extra_runner_options
+        )
```
Comment (linusg) [inline main.py:280]:
```suggestion
        self.extra_runner_options = extra_runner_options or []
```

**Review by linusg — COMMENTED** [review]
Please update the readme as part of the last two commits :)

File: main.py (hunk 6 above)
Comment (davidot) [reply main.py]: I really don't write enough python, these are such nice short hands!

**Comment by linusg** [comment]
Thanks!

**Pull request merged**